src/crystal/project/local_dependency.cr

```diff
@@ -0,0 +1,50 @@
+module Crystal
+  class LocalDependency < Dependency
+    def initialize(@path, name = nil)
+      unless @path =~ /(.*\/)*(.*)/
+        raise ProjectError.new("Invalid path name: #{path}")
+      end
+
+      @name = name || $2
+
+      unless @name
```

Comment:
What about moving this `unless` and `case` to the `Dependency` class? Apart from variable names it looks the same for local and github, and if this changes for some reason, we can then just change it in one place.